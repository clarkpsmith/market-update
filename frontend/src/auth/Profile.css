.Profile {
  text-align: left;
  color: white;
  display: flex;
  justify-content: center;
  background-color: black;
  /* height: 100vh; */
  border-radius: 1px;
}
