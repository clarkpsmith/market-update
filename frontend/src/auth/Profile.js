import React, { useState, useContext } from "react";
import "../common/SearchForm.css";
import UserContext from "../common/UserContext";
import Alert from "../helpers/Alert";
import { Button, Form, FormGroup } from "reactstrap";
import "./SignUp.css";
import { useHistory } from "react-router-dom";
import { useSelector } from "react-redux";
import ChaseLoading from "../chaseloading/ChaseLoading";

const Profile = () => {
  const { updateProfile } = useContext(UserContext);
  const currentUser = useSelector((store) => store.currentUser);
  const [formErrors, setFormErrors] = useState(null);
  const history = useHistory();
  const [loading, setLoading] = useState(false);

  function handleDelete() {
    history.push("/delete");
  }
  const initialData = {
    username: currentUser.username,
    password: "",
    firstName: currentUser.firstName,
    lastName: currentUser.lastName,
    email: currentUser.email,
  };
  const [formData, setFormData] = useState(() => initialData);

  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData((data) => ({ ...data, [name]: value }));
  }

  const handleSubmit = async (evt) => {
    evt.preventDefault();
    setLoading(true);
    const response = await updateProfile(formData);
    if (response.success) {
      history.push("/watchlist");
    } else {
      setFormErrors(response.err);
      setLoading(false);
    }
  };
  if (loading) {
    return (
      <div className="Login">
        <ChaseLoading />
      </div>
    );
  }
  return (
    <div className="SignUp">
      <div className="SignUp-block">
        <h2 className="SignUp-title">User Profile</h2>
        <Form className="SignUp-form" onSubmit={handleSubmit}>
          <FormGroup>
            <label htmlFor="username">Username</label>
            <br />
            <div>{formData.username}</div>
          </FormGroup>

          <FormGroup>
            <label htmlFor="firstName">First Name</label>
            <br />
            <input
              className="SignUp-input"
              type="text"
              name="firstName"
              id="firstName"
              placeholder={formData.firstName}
              value={FormData.firstName}
              onChange={handleChange}
            />
          </FormGroup>

          <FormGroup>
            <label htmlFor="lastName">Last Name</label>
            <br />
            <input
              className="SignUp-input"
              type="text"
              name="lastName"
              id="lastName"
              placeholder={formData.lastName}
              value={FormData.lastName}
              onChange={handleChange}
            />
          </FormGroup>
          <FormGroup>
            <label htmlFor="email">Email</label>
            <br />
            <input
              className="SignUp-input"
              type="text"
              name="email"
              id="email"
              placeholder={formData.email}
              value={FormData.email}
              onChange={handleChange}
            />
          </FormGroup>
          <FormGroup>
            <label htmlFor="password">Confirm password to make changes: </label>
            <br />
            <input
              className="SignUp-input"
              type="password"
              name="password"
              id="password"
              value={FormData.password}
              onChange={handleChange}
              required
            />
            <br />
          </FormGroup>
          {formErrors ? <Alert type="danger" messages={formErrors} /> : null}
          <Button size="md" text-align="center" className="SignUp-btn">
            Save Changes
          </Button>
          <br />
          <br />
        </Form>
        <div className="delete-profile">
          <br />
          <br />
          <Button color="danger" onClick={handleDelete}>
            Delete Profile
          </Button>
        </div>
      </div>
    </div>
  );
};
export default Profile;
