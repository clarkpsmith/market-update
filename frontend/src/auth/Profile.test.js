import React from "react";
import { render } from "@testing-library/react";
import configureMockStore from "redux-mock-store";
import Profile from "./Profile";
import { tslaData } from "../testValues";
const mockStore = configureMockStore();
import { Provider, useDispatch } from "react-redux";
import UserContext from "../common/UserContext";

const store = mockStore({
  currentUser: {
    firstName: "test",
    lastName: "user",
    email: "testuser@gmail.com",
    username: "testuser",
    watchlist: ["tsla"],
  },
  watchlistData: [tslaData.data.quoteResponse.result[0]],
  searchTickerData: tslaData.data.quoteResponse.result[0],
});

async function updateProfile(formData) {
  const dispatch = useDispatch();
  try {
    const res = await MarketUpdateApi.updateProfile(formData);

    dispatch({ type: UPDATE_PROFILE, currentUser: res.user });
    return { success: true };
  } catch (err) {
    console.error("Update Profile Failed", err);
    return { success: false, err };
  }
}

it("renders without crashing", () => {
  render(
    <Provider store={store}>
      <UserContext.Provider value={{ updateProfile }}>
        <Profile />
      </UserContext.Provider>
    </Provider>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <UserContext.Provider value={{ updateProfile }}>
        <Profile />
      </UserContext.Provider>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
