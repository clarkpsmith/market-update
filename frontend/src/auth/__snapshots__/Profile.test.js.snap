// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SignUp"
  >
    <div
      class="SignUp-block"
    >
      <h2
        class="SignUp-title"
      >
        User Profile
      </h2>
      <form
        class="SignUp-form"
      >
        <div
          class="form-group"
        >
          <label
            for="username"
          >
            Username
          </label>
          <br />
          <div>
            testuser
          </div>
        </div>
        <div
          class="form-group"
        >
          <label
            for="firstName"
          >
            First Name
          </label>
          <br />
          <input
            class="SignUp-input"
            id="firstName"
            name="firstName"
            placeholder="test"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            for="lastName"
          >
            Last Name
          </label>
          <br />
          <input
            class="SignUp-input"
            id="lastName"
            name="lastName"
            placeholder="user"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            for="email"
          >
            Email
          </label>
          <br />
          <input
            class="SignUp-input"
            id="email"
            name="email"
            placeholder="testuser@gmail.com"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            for="password"
          >
            Confirm password to make changes: 
          </label>
          <br />
          <input
            class="SignUp-input"
            id="password"
            name="password"
            required=""
            type="password"
            value=""
          />
          <br />
        </div>
        <button
          class="SignUp-btn btn btn-secondary btn-md"
          text-align="center"
        >
          Save Changes
        </button>
        <br />
        <br />
      </form>
      <div
        class="delete-profile"
      >
        <br />
        <br />
        <button
          class="btn btn-danger"
          type="button"
        >
          Delete Profile
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
